pkg/render/intrusion_detection.go

```diff
@@ -85,13 +87,16 @@ type intrusionDetectionComponent struct {
 	esLicenseType     ElasticsearchLicenseType
 	jobInstallerImage string
 	controllerImage   string
+	managedCluster    bool
 }
 
 func (c *intrusionDetectionComponent) ResolveImages(is *operator.ImageSet) error {
 	reg := c.installation.Registry
 	path := c.installation.ImagePath
 	var err error
-	c.jobInstallerImage, err = components.GetReference(components.ComponentElasticTseeInstaller, reg, path, is)
+	if !c.managedCluster {
+		c.jobInstallerImage, err = components.GetReference(components.ComponentElasticTseeInstaller, reg, path, is)
+	}
 	errMsgs := []string{}
 	if err != nil {
```

Comment:
Would prefer to move this error handling inside the new `if !c.managedCluster` block above so that it's more obvious where the error is coming from.

Reply:
Put up a PR for this https://github.com/tigera/operator/pull/1183.